src/meili/traffic_segment_matcher.cc

```diff
@@ -275,6 +278,16 @@ std::vector<traffic_segment_t> TrafficSegmentMatcher::form_segments(const std::l
       if(left->edge_distance > segment->begin_percent_)
         left = std::prev(left);
 
+      //move the right marker right until its adjacent to the segment end
+      right = std::find_if(right, markers.cend(), [&segment](const interpolation_t& mark) {
+        return mark.edge == segment.end_edge && mark.edge_distance >= segment->end_percent_;
+      });
+
+      //skip any segments composed entirely of transition edges (should only be one edge really)
+      //they should have no valid segment id, be marked internal and also have no way ids
+      if(!segment->segment_id_.Is_Valid() && segment.internal && segment.way_ids.empty())
```

Comment:
the second part of this trick is to elide segments who have no way ids given that they must be transition edges